CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Contribution Guidelines
+We welcome feedback, bug reports, and pull requests!
+For pull requests (PRs), please stick to the following guidelines:
+* Before submitting a PR, verify that an issue exists that describes the bug fix or feature you want to contribute. If there's no issue yet, please create one.
+* Fork the repo on your GitHub user account, make code changes there, and then create a PR against the `main` branch.
+* Add tests for any new features or bug fixes. Ideally, each PR increases the test coverage. We have a quality gate that checks.
+* Follow the existing code style (follow [Effective Go](https://go.dev/doc/effective_go) where no obvious convention exists).
+* Use the pull request template provided, and fill in all the details you can.
+* Document newly introduced methods and classes with godoc-style comments, and add inline comments to code that is not self-documenting.
+* Separate unrelated changes into multiple PRs.
+* Commit messages should follow the conventions outlined at https://www.conventionalcommits.org/.
+### Hooks
```

Comment:
```suggestion

### Hooks
```

Reply:
Do we just want to call it `Pre-Commit Hooks`?

Reply:
Originally was thinking we can call it hooks, as I've found it useful to only run tests on pre-push. And so it would really be "pre-commit and pre-push hooks". 

I guess this is only useful for repos where tests take a long time, but might be useful to be a bit more generic here, so that we have freedom to add this hook as a pre-push, if we want to.